File: java/src/chapter_2/Problem_2.java
package chapter_2;

import java.util.Scanner;

public class Problem_2 {

  public static void main(String[] args) {
    // Enunciado: Hallar el cociente y residuo de dos numeros enteros

    // Declaraciones
    Scanner myscan = new Scanner(System.in);
    int num1, num2;
    float cociente, residuo;

    // Inputs
    System.out.println("Ingrese un primer numero");
    num1 = myscan.nextInt();
    System.out.println("Ingrese un segundo numero");
    num2 = myscan.nextInt();

    // Proceso
    // Se hace uso de un casting temporal a float
    cociente = (float) num1 / (float) num2;
    residuo = num1 % num2;

    // Salida
    // Es posible decir desde printf cuandos decimales queremos
    System.out.printf("El cociente de la operacion es %.2f. El residuo es %.2f", cociente, residuo);
    myscan.close();

    // FIN
  }

}

File: ruby/chapter_2/problem_2.rb
# Enunciado: Hallar el cociente y residuo de dos numeros enteros

# Declaraciones explicitas no necesarias en Ruby
num1, num2 = 0
cociente, residuo = 0.0

# Inputs
puts "Ingrese un primer numero"
num1 = gets().chomp().to_i

puts "Ingrese un segundo numero"
num2 = gets().chomp().to_i

# Proceso
cociente = (num1.to_f / num2.to_f)
residuo = num1 % num2

# Salida
puts "El cociente de la operacion es #{cociente}. El residuo es #{residuo}."
